Gems/ROS2/Code/Source/SimulationUtils/FollowingCameraComponent.cpp

```diff
@@ -170,93 +194,323 @@ namespace ROS2
     }
 
-    AZ::Vector3 FollowingCameraComponent::AverageVector(const AZStd::deque<AZStd::pair<AZ::Vector3, float>>& buffer) const
+    AZ::Vector3 FollowingCameraComponent::WeightedAverageVector(const AZStd::deque<AZStd::pair<AZ::Vector3, float>>& buffer) const
     {
-        AZ::Vector3 sum{ 0 };
-        float normalization{ 0 };
-        for (const auto& p : buffer)
+        AZ::Vector3 weightedSum = AZ::Vector3::CreateZero();
+        float totalWeight = 0.0f;
+        float currentWeight = 1.0f; // Start with a weight of 1 for the oldest entry
+        float weightIncreaseFactor = m_configuration.m_smoothFactor; // Increase each subsequent weight by 10%
+
+        for (const auto& entry : buffer)
+        {
+            weightedSum += entry.first * currentWeight; // Apply current weight to the vector
+            totalWeight += currentWeight; // Accumulate total weight for normalization
+            currentWeight *= weightIncreaseFactor; // Increase weight for the next, more recent entry
+        }
+
+        if (totalWeight > 0.0f)
         {
-            sum += p.first * p.second;
-            normalization += p.second;
+            return weightedSum / totalWeight; // Normalize weighted sum by total weight
         }
-        return sum / normalization;
+
+        return AZ::Vector3::CreateZero(); // Return zero vector if buffer is empty or totalWeight is somehow zero
     }
 
     AZ::Vector3 FollowingCameraComponent::SmoothTranslation() const
     {
-        return AverageVector(m_lastTranslationsBuffer);
+        // Use the modified weighted average calculation for smoothing
+        return WeightedAverageVector(m_lastTranslationsBuffer);
     }
 
     AZ::Quaternion FollowingCameraComponent::SmoothRotation() const
     {
-        AZ::Quaternion q = m_lastRotationsBuffer.front().first;
-        for (int i = 1; i < m_lastRotationsBuffer.size(); i++)
+        if (m_lastRotationsBuffer.empty())
         {
-            q = q.Slerp(m_lastRotationsBuffer[i].first, m_lastRotationsBuffer[i].second);
+            return AZ::Quaternion::CreateIdentity(); // Return identity quaternion if the buffer is empty
         }
-        return q;
+
+        // Start with the oldest rotation in the buffer
+        AZ::Quaternion smoothedRotation = m_lastRotationsBuffer.front().first;
+        float totalWeight = 0.0f;
+        float currentWeight = 1.0f; // Initial weight
+        float weightIncreaseFactor = m_configuration.m_smoothFactor; // Determines how much more influence each subsequent rotation has
+
+        for (size_t i = 1; i < m_lastRotationsBuffer.size(); ++i)
+        {
+            totalWeight += currentWeight;
+            float t = currentWeight / totalWeight; // Calculate interpolation factor
+
+            // Progressively slerp towards each newer rotation, with increasing influence
+            smoothedRotation = smoothedRotation.Slerp(m_lastRotationsBuffer[i].first, t);
```

Comment:
I do not understand this logic.

Reply:
Let me share the link to the [matlab docs](https://www.mathworks.com/help/nav/ug/lowpass-filter-orientation-using-quaternion-slerp.html). Spherical linear interpolation.

Reply:
Please add a proper description of the algorithm and assert on t > 0 and t < 1.